### ISSUE TITLE
SwipeableTabBarController freezes after a few swipes

### ISSUE THREAD

**Issue by filip-jaksic** [marcosgriselli/SwipeableTabBarController#48]
Hello,

I was checking out your framework and decided to try out the demo project. Unfortunately, it freezes every time after a few swipes; no errors are displayed, it just stops responding. Not sure if's me or something, but I downloaded it straight from here (v1.3.0) and have the latest xCode (9.4.1).
   ¯\_(ツ)_/¯

Here's a short video of the problem:  https://streamable.com/6bd2o

**Comment by cdeck95** [comment]
Same thing happens for me. Only thing I did was add it to my project with cocoapods and then select the class in storyboard. Freezes after a few

**Comment by marcosgriselli** [comment]
Is it happening on a real device as well? I had multiple issues with the simulator navigation. I'll be reviewing the transitions next weekend. I'll create a milestone for this issue.

**Comment by cdeck95** [comment]
Yes, i never tested on a simulator, just put it on my phone. Not sure if this matters, but the tab bar controller has 6 view controllers connected to it, so the last couple show up in the “more” section

**Comment by marcosgriselli** [comment]
VC count shouldn't be a problem. I'll take a look at it.

**Comment by marcosgriselli** [comment]
@FilipJaksic @cdeck95 I think I nailed the solution on this new branch: https://github.com/marcosgriselli/SwipeableTabBarController/tree/fix/48_freeze 

It's a strange issue to debug but I've been swiping and tapping back and forth in both simulator an physical device with no freezes.

**Comment by cdeck95** [comment]
Thanks! Can you pull that into Master? I want to implement it into my app. Is there a way with cocoapods to import from a specific branch instead of master? lol

**Comment by marcosgriselli** [comment]
You should be able to install the specific branch using:

```ruby
pod 'SwipeableTabBarController', :git => 'https://github.com/marcosgriselli/SwipeableTabBarController.git', :branch => 'fix/48_freeze'
```
If it works on your app I’ll merge it to master and submit a new version.

**Comment by marcosgriselli** [comment]
@cdeck95 any luck with it?

**Comment by cdeck95** [comment]
Hey I didn’t have any luck with the fix. It didn’t work for me, but it could have been a mid configuration on my part. I am going to try it in a brand new project tonight and keep you updated.

**Comment by cdeck95** [comment]
No problems in a new project. So must be a misconfiguration on my end because it still doesn’t work on my app. 

Things I have in my app and my guesses right now:

-Tab bar controller with 6 tabs. 
-Each tab has a navigation controller (should tab bar controller be inside one navigation controller? Idk I need to check)
-main tab has a collection view that swipes horizontally (problem with different swipe gestures?)
-second tab is a camera, so opens a camera. Maybe having a camera cover the screen doesn’t allow for swiping, not sure. But it crashes before getting to this controller
-nothing else matters because I can’t even swipe right to the second view controller (the camera)

**Comment by cdeck95** [comment]
Did some digging:

The swiping worked in my app after i got rid of the view controller than was basically just a camera, and replaced it with another view controller.

Also another important note: It seems to only work when every view controller is embedded in a navigation controller (which is the standard practice for working with both anyway)

TL;DR - Not something that is your fault, but my app seems to be incompatible with this library. But this can be closed since this issue is specifically for me!

**Comment by marcosgriselli** [comment]
Hey @cdeck95 thanks for taking the time for digging into this! I guess I should've asked about your view hierarchy in the first place 😅. Haven't really tried it with a view controller that only displays a camera and yes horizontal collections views conflict with the gesture and there's not much I can do about that one 😕. 

I'll evaluate solutions as I think you're not the first one trying to display a camera or horizontal collection view. I'll close this issue and handle the updates on separate ones.

**Comment by cdeck95** [comment]
Sounds good! Here is the code I use to launch the camera in my app to get you started...

```
import UIKit
import AVFoundation

class ScannerViewController: UIViewController, AVCaptureMetadataOutputObjectsDelegate {
    
    var captureSession: AVCaptureSession!
    var previewLayer: AVCaptureVideoPreviewLayer!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        captureSession = AVCaptureSession()
        self.navigationController?.navigationBar.isHidden  = true

        guard let videoCaptureDevice = AVCaptureDevice.default(for: .video) else { return }
        let videoInput: AVCaptureDeviceInput
        
        do {
            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
        } catch {
            return
        }
        
        if (captureSession.canAddInput(videoInput)) {
            captureSession.addInput(videoInput)
        } else {
            failed()
            return
        }
        
        let metadataOutput = AVCaptureMetadataOutput()
        
        if (captureSession.canAddOutput(metadataOutput)) {
            captureSession.addOutput(metadataOutput)
            
            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
            metadataOutput.metadataObjectTypes = [.qr]
        } else {
            failed()
            return
        }
        
        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
        previewLayer.frame = view.layer.bounds
        previewLayer.videoGravity = .resizeAspectFill
        view.layer.addSublayer(previewLayer)
        
        captureSession.startRunning()
    }
    
    func failed() {
        let ac = UIAlertController(title: "Scanning not supported", message: "Your device does not support scanning a code from an item. Please use a device with a camera.", preferredStyle: .alert)
        ac.addAction(UIAlertAction(title: "OK", style: .default))
        present(ac, animated: true)
        captureSession = nil
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        if (captureSession?.isRunning == false) {
            captureSession.startRunning()
        }
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        
        if (captureSession?.isRunning == true) {
            captureSession.stopRunning()
        }
    }
```
